easy-ui-react/src/Calendar/CalendarGrid.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { useCalendarGrid, useLocale } from "react-aria";
+import { getWeeksInMonth } from "@internationalized/date";
+import { CalendarState } from "@react-stately/calendar";
+import { CalendarDate, isSameMonth } from "@internationalized/date";
+import { Text } from "../Text";
+import { CalendarCell } from "./CalendarCell";
+import styles from "./CalendarGrid.module.scss";
+
+export type CalendarGridProps = {
+  state: CalendarState;
+  startDate?: CalendarDate;
+  endDate?: CalendarDate;
+  weekdayStyle?: "narrow" | "short" | "long";
+  showOutsideDays?: boolean;
+};
+
+export function CalendarGrid({ state, ...props }: CalendarGridProps) {
+  const { locale } = useLocale();
+  const { showOutsideDays } = props;
+  const { gridProps, headerProps, weekDays } = useCalendarGrid(props, state);
+
+  const firstDateOfMonth = state.visibleRange.start;
+
+  // Get the number of weeks in the month so we can render the proper number of rows.
+  const weeksInMonth = getWeeksInMonth(firstDateOfMonth, locale);
+
+  return (
+    <table {...gridProps} className={styles.CalendarGrid}>
+      <thead {...headerProps} className={styles.CalendarGridHeader}>
+        <tr>
+          {weekDays.map((day, index) => (
+            <th key={index}>
+              <Text variant="overline" color="neutral.000">
+                {day}
+              </Text>
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {[...new Array(weeksInMonth).keys()].map((weekIndex) => (
+          <tr key={weekIndex}>
+            {state
+              .getDatesInWeek(weekIndex)
+              .map((date: CalendarDate | null, i: number) =>
+                date &&
```

Comment:
I tested it, and it seems to always have 7 dates in a week. Right now, `<td />` renders when the date is outside of the current month, and `showDaysOutsideCurrentMonth` returns `false`

Reply:
i guess i'm wondering about the `date &&` line. from my initial understanding of it, it seems like it's not doing anything since `date` will always be truthy. but it's possible i'm not understanding every case

Reply:
It seems like few calendar systems don't have 7 days in a week (Javanese calendar), and checking `date` is to capture that.